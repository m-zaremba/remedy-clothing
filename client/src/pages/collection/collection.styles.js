import styled from 'styled-components';

export const CollectionPageContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const CollectionTitle = styled.h2`
  font-size: 28px;
  margin: 0 0 25px 0;
  text-align: left;
  text-transform: uppercase;
`;

export const CollectionItemsContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 10px;

  @media screen and (max-width: 800px) {
    display: grid;
    grid-template-columns: 1fr;
    padding: 10px;
    margin: 0 auto;
  }
`;
