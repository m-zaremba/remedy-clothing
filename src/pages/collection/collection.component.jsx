import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import {
  CollectionPageContainer,
  CollectionTitle,
  CollectionItemsContainer,
} from './collection.styles';

import { selectCollection } from '../../redux/shop/shop.selectors';

import CollectionItem from '../../components/collection-item/collection-item.component';

const CollectionPage = ({ collection }) => {
  const { title, items } = collection;
  return (
    <CollectionPageContainer>
      <CollectionTitle>{title}</CollectionTitle>
      <CollectionItemsContainer>
        {items.map((item) => (
          <CollectionItem key={item.id} item={item} />
        ))}
      </CollectionItemsContainer>
    </CollectionPageContainer>
  );
};

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollection(ownProps.match.params.collectionId)(state),
});

CollectionPage.propTypes = {
  collection: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    routeName: PropTypes.string,
    items: PropTypes.array,
  }),
};

export default connect(mapStateToProps)(CollectionPage);
