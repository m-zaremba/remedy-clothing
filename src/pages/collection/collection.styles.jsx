import styled from 'styled-components';

export const CollectionPageContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const CollectionTitle = styled.h2`
  font-size: 28px;
  margin: 0 0 25px 0;
  text-align: left;
  text-transform: uppercase;
`;

export const CollectionItemsContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 10px;
`;
