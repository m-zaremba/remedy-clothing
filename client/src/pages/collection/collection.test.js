import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './collection.component';
import CollectionItem from '../../components/collection-item/collection-item.component';

describe('CollectionPage', () => {
  let wrapper;
  const mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];
  beforeEach(() => {
    const mockCollection = {
      items: mockItems,
      title: 'test collections',
    };

    wrapper = shallow(<CollectionPage collection={mockCollection} />);
  });

  it('should render the CollectionPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the same number of CollectionItems as mockItems array', () => {
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
  });
});
