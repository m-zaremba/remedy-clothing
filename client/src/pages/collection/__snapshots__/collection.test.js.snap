// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage should render the CollectionPage component 1`] = `
<styled.div>
  <styled.h2>
    test collections
  </styled.h2>
  <styled.div>
    <Connect(CollectionItem)
      item={
        Object {
          "id": 1,
        }
      }
      key="1"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 2,
        }
      }
      key="2"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 3,
        }
      }
      key="3"
    />
  </styled.div>
</styled.div>
`;
